src/storage-controller/src/lib.rs

```diff
@@ -2039,17 +2123,17 @@ where
             .await
             .expect("stash operation must succeed");
 
+        let txns_id = PERSIST_TXNS_SHARD
```

Comment:
A notable effect of this commit as it currently stands is that this will create the txns shard and use a bit of the system even if the flag has never been on. Looking at this commit again, I think we may want to instead use wether `PERSIST_TXNS_SHARD` has ever been created in the storage controller stash as knowledge of if the feature has even been turned on before. That plus `enable_persist_txn_tables` should let us be lazy about the init so this is a complete no-op in prod

Reply:
@jkosh44 I tried to swap this to a `peek_key_one` call when the flag is off and am getting this. Any pointers?

> could not get txns shard id: StashError { inner: PeekSinceUpper("collection 920972787545243649 since {-9223372036854775808} is not less than upper {-9223372036854775808}") }

Reply:
Yeah, stash collections cannot be read until they've been initialized (i.e. written to once). Usually we'll execute an empty write for new collections. I'll try and dig up some examples.

Reply:
Done!